### PR TITLE
Snode communication retries and logging improvement

### PR DESCRIPTION
**Pull request by neuroscr**
- `sendToProxy()` handle non base64 responses from snode
- `getSwarmNodes()` - Hit multiple random nodes and merge lists
- minor optimization: `refreshSwarmNodesForPubKey()` returns filtered nodes to avoid hitting sqlite again
- `initialiseRandomPool()` retries up to 3 times before failing
- `startLongPolling()` logs if pulling from blockchain fails
- include server ip/port in logging for additional manual testing
- log some unlogged exception

basically improves on:
`Loki retrieve messages:  Illegal str: Length not a multiple of 4`
`Loki retrieve messages:  Illegal character code: 32`
to actually give actionable information

### CODE REVIEW

File: js/modules/loki_rpc.js [hunk 1]
```diff
@@ -67,20 +67,42 @@ const sendToProxy = async (options = {}, targetNode) => {
...
-  const plaintextBuffer = await window.libloki.crypto.DHDecrypt(
-    symmetricKey,
-    ciphertextBuffer
-  );
+  const ciphertext = await response.text();
+  let plaintext
+  try {
+    // removed the following part to match the check we have in loki_app_dot_net_api.js
+    // ; not done syncing;
+    if (ciphertext.match(/Service node is not ready: not in any swarm/)) {
```
Comment (Mikunj) [inline js/modules/loki_rpc.js]: Could we potentially check for a http code instead of matching exact text?

File: js/modules/loki_snode_api.js [hunk 2]
```diff
@@ -107,20 +119,23 @@ class LokiSnodeAPI {
   }
 
   async updateSwarmNodes(pubKey, newNodes) {
+    let filteredNodes = [];
     try {
-      const filteredNodes = newNodes.filter(snode => snode.ip !== '0.0.0.0');
+      filteredNodes = newNodes.filter(snode => snode.ip !== '0.0.0.0');
```
Comment (Mikunj) [inline js/modules/loki_snode_api.js]: You can just return this within the try statement

File: js/modules/loki_rpc.js (hunk 1 above)
Comment (neuroscr) [reply js/modules/loki_rpc.js]: Looks like it so far
